### PR TITLE
added lesson 4 assignment 2, mailroom - part2

### PR DESCRIPTION
**Pull request by TacSPx**
Please see my completed lesson 4 assignment 2, mailroom - part2:
mailroompart2.py

### CODE REVIEW

**Comment by TacSPx** [comment]
Please see my completed lesson 5, exercise, exceptions:
except_exercise.py
except_test.py


Not sure why chris_delapena files are included in my post.  I think we posted around the same time today and I could not remove his, so I just posted.

File: students/TacSPx/lesson04/mailroompart2.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+        if (donor_name.lower() == 'list'):
+            print("\nDONOR REGISTRY:")
+            donor_sort = (sorted(donor_info.keys()))  # used .keys() to sort on key
+            for key in donor_sort:
+                print(key)
+            input("Press Enter to Continue")
+            print()
+            continue
+
+        if (donor_name.lower() != 'list'):
```
Comment (natasha-aleksandrova) [inline students/TacSPx/lesson04/mailroompart2.py:81]: this should be an `else` statement

File: students/TacSPx/lesson04/mailroompart2.py [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+    while True:
+        donor_name = input("Type 'List' to see all names in Database or\nType First and Last Name: ").title()
+        if (donor_name.lower() == 'list'):
+            print("\nDONOR REGISTRY:")
+            donor_sort = (sorted(donor_info.keys()))  # used .keys() to sort on key
+            for key in donor_sort:
+                print(key)
+            input("Press Enter to Continue")
+            print()
+            continue
```
Comment (natasha-aleksandrova) [inline students/TacSPx/lesson04/mailroompart2.py:79]: you don't need a continue here - there is no other code to run after this block

File: students/TacSPx/lesson04/mailroompart2.py [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+
+        if (donor_name.lower() != 'list'):
+            donor_amount = int(input("Enter Amount of Donation: $"))
+            # if name is in dict append dollar amount to donors bucket
+            for i in donor_info:
```
Comment (natasha-aleksandrova) [inline students/TacSPx/lesson04/mailroompart2.py:84]: looping over a dictionary to find the key is incorrect - you can access and check they key by either index access or via get() method.

File: students/TacSPx/lesson04/mailroompart2.py [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+
+        # calcs in the dictionary
+        donor_sum = dict([(key, sum(values)) for key, values in donor_info.items()])
+        donor_len = dict([(key, len(values)) for key, values in donor_info.items()])
+        donor_avg = dict([(key, sum(values) / len(values)) for key, values in donor_info.items()])
+        # merge
+        for key in set(list(donor_sum.keys()) + list(donor_len.keys()) + list(donor_avg.keys())):
+            donor_merge.setdefault(key, []).append(donor_sum[key])
+            donor_merge.setdefault(key, []).append(donor_len[key])
+            donor_merge.setdefault(key, []).append(donor_avg[key])
```
Comment (natasha-aleksandrova) [inline students/TacSPx/lesson04/mailroompart2.py:152]: not sure on the purpose of the above 3 dictionaries and merging them - you shouldn't need any of that

pass donor_info to sorted() function and sort on the sum

File: students/TacSPx/lesson04/mailroompart2.py [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+                #print(donor_name, donor_amount) # for testing
+                all_letters = letter(donor_name, donor_amount)
+                # creates the thank you letter in a text file
+                with open(donor_name + ".txt", "w") as f: # use donor name as file name
+                    f.write(all_letters)
+            input("\nCompleted creating Letters! Please check program directory for files \nPress Enter to Continue!")
+            break
+
+        elif choice.lower() != 'y':
+            main()
```
Comment (natasha-aleksandrova) [inline students/TacSPx/lesson04/mailroompart2.py:185]: if you want to return to main menu then have a `return` here - shouldn't call main() loop directly

File: students/TacSPx/lesson04/mailroompart2.py [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+    Will exit the program
+    :return:
+    '''
+    while True:
+        exit_choice = input("Would you like to Exit?\nEnter 'y' or 'n': ")
+        if exit_choice.lower() == 'y':
+            print("\nExiting!")
+            sys.exit()
+        elif exit_choice.lower() == 'n':
+            main()
```
Comment (natasha-aleksandrova) [inline students/TacSPx/lesson04/mailroompart2.py:202]: same here - don't call main directly

**Pull request merged**